cpp/mrc/src/internal/data_plane/data_plane_resources.cpp

```diff
@@ -363,6 +364,58 @@ std::shared_ptr<ucxx::Request> DataPlaneResources2::am_recv_async(std::shared_pt
     return request;
 }
 
+uint64_t DataPlaneResources2::get_next_object_id()
+{
+    auto object_id = m_next_object_id++;
+    while (m_remote_descriptor_by_id.find(object_id) != m_remote_descriptor_by_id.end())
+    {
+        object_id = m_next_object_id++;
+    }
+
+    return object_id;
```

Comment:
Can we just replace this with an atomic? We dont really care what the ID is, just that its unique.

Reply:
Yes, using an atomic here makes sense. I'm not sure I understand your comment about uniqueness though, perhaps it's because the `find` which I added there to ensure we never reassign an ID that's in use (i.e., if we go over maximum number of IDs), which is probably unlikely given we use `uint64_t`, I can remove that check if you think we're reasonably confident we'll never go beyond the `uint64_t` limit.

Reply:
I think its unlikely we will send more than 18,446,744,073,709,551,615 messages.